grep/src/literals.rs

```diff
@@ -233,9 +233,9 @@ fn count_unicode_class(cls: &hir::ClassUnicode) -> u32 {
     cls.iter().map(|r| 1 + (r.end() as u32 - r.start() as u32)).sum()
 }
 
-/// Return the number of bytes in the given class.
+/// Return the number of bytes in the givep class.
 fn count_byte_class(cls: &hir::ClassBytes) -> u32 {
-    cls.iter().map(|r| 1 + (r.end() as u32 - r.start() as u32)).sum()
+    cls.iter().map(|r| 1 + (u32::from(r.end()) - u32::from(r.start()))).sum()
```

Comment:
I don't understand this change. Can you justify it?

Reply:
IIUC, `as` can perform lossy conversions, whereas `From` is only implemented for lossless ones, so if the type of `r.end()` and `r.start()` changed to something like `u64` it would truncate. In this case that would almost certainly never happen, so it's unlikely to be a real problem here, but I think the idea is that it's usually the safer default so clippy lints against it.

Reply:
`ClassByteRange` is always going to be a byte range. But I don't feel that strongly. Happy to keep this.